keyvi/include/keyvi/vector/vector_file.h

```diff
@@ -0,0 +1,147 @@
+/* * keyvi - A key value store.
+ *
+ * Copyright 2018   Narek Gharibyan<narekgharibyan@gmail.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ *  vector_file.h
+ *
+ *  Created on: March 17, 2018
+ *      Author: Narek Gharibyan <narekgharibyan@gmail.com>
+ */
+
+#ifndef KEYVI_VECTOR_VECTOR_FILE_H_
+#define KEYVI_VECTOR_VECTOR_FILE_H_
+
+#include <exception>
+#include <fstream>
+#include <memory>
+#include <string>
+
+#include <boost/property_tree/ptree.hpp>
+
+#include "dictionary/fsa/internal/value_store_factory.h"
+#include "util/serialization_utils.h"
+#include "vector/types.h"
+
+static const char KEYVI_VECTOR_BEGIN[] = "KEYVI_VECTOR_BEGIN";
+static const size_t KEYVI_VECTOR_BEGIN_LEN = 18;
+static const char KEYVI_VECTOR_END[] = "KEYVI_VECTOR_END";
+static const size_t KEYVI_VECTOR_END_LEN = 16;
+static const char MANIFEST[] = "manifest";
+static const char SIZE[] = "size";
```

Comment:
as this is a header implementation, I suggest to be careful on naming, SIZE and MANIFEST sound very generic.